### ISSUE TITLE
Contiguously coloured field disappears when finding the last character that ends a corlorised field.

### ISSUE THREAD

**Issue by Ma-XX-oN** [gwsw/less#265]
For the following command:
```
echo Hello there | grep --color=always Hello | less -R
```
If you were to search for the letter 'o', the word 'Hello' disappears.  This happens when a search finds the end of any contagiously coloured field.  I.e. 'lo', 'llo', 'ello', 'Hello' all cause 'Hello' to disappear.

**Comment by gwsw** [comment]
I believe this was fixed in less-596. Can you try the current beta (less-603) and see if you still see the bug?

**Comment by Ma-XX-oN** [comment]
Sure.  How do I do that?  Do I make a build or something?

**Comment by Ma-XX-oN** [comment]
Oh, and are you guys supporting searching across colour field boundaries?  Because it does some messed up stuff.

**Comment by Ma-XX-oN** [comment]
Tried to build, but `sh configure --with-regex=none` failed because it couldn't find the terminal libraries.  I dled the latest ncurses lib using the `pacman -S mingw-w64-i686-ncurses` command (I'm using MSYS2) but it didn't work.  Any ideas.

**Comment by Ma-XX-oN** [comment]
Nope, still an issue.  My command:
```
echo Why hello there | grep --color=always hello | less -R
```
Searching for 'll':
![image](https://user-images.githubusercontent.com/10625935/169061037-bd1dd884-267d-4ac3-9df4-115aac89a98c.png)
Searching for 'o' or 'lo':
![image](https://user-images.githubusercontent.com/10625935/169061148-62b847ec-10f5-4acb-b908-91ff6a74d0fa.png)
Searching for 'lo t':
![image](https://user-images.githubusercontent.com/10625935/169061239-b701369c-c39c-448f-99cf-19361377bcb1.png)
Searching for 'y h':
![image](https://user-images.githubusercontent.com/10625935/169061614-dd1eb878-e33d-4a00-9d61-bd624cc3d58d.png)
And then after updating the terminal with C-l:
![image](https://user-images.githubusercontent.com/10625935/169061724-f43fcfac-352d-4331-b6f9-0b4ae0086d21.png)

**Comment by Ma-XX-oN** [comment]
Well it does work better. However, I noticed that the inverted highlight box has only the colour of where the selection started.
My command:
```
echo Why hello there | grep --color=always hello | ./less -R
```
Select `y he` will show the selected as black on white:
![image](https://user-images.githubusercontent.com/10625935/169063205-72f8e661-9a8b-487b-af4a-04d04aa61445.png)

Where as sellect `llo th` will show the selected as black on red:
![image](https://user-images.githubusercontent.com/10625935/169063107-b999b016-ad51-4bb8-b623-9fe0cc95a063.png)

I would have expected it to just invert the colours of each character the selection was applied to.  But definitely an improvement.

**Comment by gwsw** [comment]
@Ma-XX-oN This last issue should be fixed in less-608 (c7ed204446892814c3db47d3cfabc598371700f6). Let me know if you see any more problems.

**Comment by alejandro-colomar** [comment]
I think this same issue is the one I faced lately:

<https://lists.gnu.org/archive/html/groff/2022-08/msg00116.html>
<https://lists.gnu.org/archive/html/groff/2022-08/msg00116.html>

I compiled less 608 from source, and I can't reproduce my issue anymore.

**Comment by Ma-XX-oN** [comment]
> @Ma-XX-oN This last issue should be fixed in less-608 ([c7ed204](https://github.com/gwsw/less/commit/c7ed204446892814c3db47d3cfabc598371700f6)). Let me know if you see any more problems.

Sorry, didn't see this.  Yes it appears to be fixed.  Thx.
